posthog-core/src/gzip.ts

```diff
@@ -0,0 +1,25 @@
+/**
+ * Older browsers and some runtimes don't support this yet
+ */
+export function isGzipSupported(): boolean {
+  return 'CompressionStream' in globalThis
+}
+
+/**
+ * Gzip a string using Compression Streams API if it's available
+ */
+export async function gzipCompress(input: string): Promise<Blob> {
+  if (!isGzipSupported()) {
+    throw new Error('Compression Streams API is not available in this runtime')
```

Comment:
we should not throw here and return nothing so we can fallback to not compressed instead of failing hard

Reply:
or wrap the whole method in a try catch so its safe to fail/log and move on

Reply:
yep this now fails silently and returns null